### ISSUE TITLE
[receiver/mongodbreceiver]: Add metric version & storage engine check before recording metrics

### ISSUE THREAD

**Issue by JonathanWamsley** [open-telemetry/opentelemetry-collector-contrib#13155]
**Describe the bug**
Partial errors are being recorded in integration tests when they don't have to be. Before metrics are recorded, the mongodb version and storage engine should be checked instead of emitting a partial error.

**Steps to reproduce**
Run integration test and check for partial errors before scrape returns.
At the end of scrape, running an integration test like 5.0 I would expect no partial errors.
```go
	errs := &scrapererror.ScrapeErrors{}
	s.collectMetrics(ctx, errs)
	if errorsCombined := errs.Combine(); errorsCombined != nil {
		panic(fmt.Sprintf("Error Amount:%#v\nErrors: %#v", errorsCombined, errorsCombined.Error()))
	}
	return s.mb.Emit(), errs.Combine()
```

**What did you expect to see?**
Expect no partial errors when running integration tests.

**What did you see instead?**
For mongodb integration test 5
```log
Error Amount:scrapererror.PartialScrapeError{error:(*multierr.multiError)(0xc0000b4420), Failed:4}
Errors: "could not find key for metric; could not find key for metric; could not find key for metric; could not find key for metric"
```

**What version did you use?**
on main, mongodb version 2.6, 3, 4 and 5

**Additional context**
These partial errors got passed silently and were fixed recently from #13109  . These partial errors will stop reporting with the correct version and storage engine checks which will be documented in the readme.

**Comment by JonathanWamsley** [comment]
I'll be working on this issue.

**Comment by JonathanWamsley** [comment]
resolved with #13329
